File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/test_100/LogicNode.java
package net.merayen.elastic.backend.logicnodes.list.test_100;

import net.merayen.elastic.backend.nodes.BaseLogicNode;
import net.merayen.elastic.backend.nodes.BaseNodeProperties;
import net.merayen.elastic.system.intercom.NodeDataMessage;

public class LogicNode extends BaseLogicNode {

	@Override
	protected void onInit() {
		createInputPort("input");
	}

	@Override
	protected void onParameterChange(BaseNodeProperties instance) {
		updateProperties(instance);
	}

	@Override
	protected void onConnect(String port) {}

	@Override
	protected void onDisconnect(String port) {}

	@Override
	protected void onRemove() {}

	@Override
	protected void onData(NodeDataMessage data) {}
}

File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/compressor_1/LogicNode.kt
package net.merayen.elastic.backend.logicnodes.list.compressor_1

import net.merayen.elastic.backend.logicnodes.Format
import net.merayen.elastic.backend.nodes.BaseLogicNode
import net.merayen.elastic.backend.nodes.BaseNodeProperties
import net.merayen.elastic.system.intercom.InputFrameData
import net.merayen.elastic.system.intercom.NodeDataMessage
import net.merayen.elastic.system.intercom.OutputFrameData

class LogicNode : BaseLogicNode() {
	override fun onData(data: NodeDataMessage) {}
	override fun onConnect(port: String?) {}
	override fun onDisconnect(port: String?) {}
	override fun onRemove() {}
	override fun onPrepareFrame(): InputFrameData = InputFrameData(id)

	override fun onFinishFrame(data: OutputFrameData?) {
		sendDataToUI(data)
	}

	override fun onParameterChange(instance: BaseNodeProperties) = updateProperties(instance)

	override fun onInit() {
		createInputPort("input")
		createInputPort("sidechain")
		createOutputPort("output", Format.AUDIO)
		createOutputPort("attenuation", Format.SIGNAL)
	}
}